tests/test_output_formatters.py

```
@@ -0,0 +1,118 @@
+        Integration test: Run the CLI with --json flag
+        This goes in a separate test file like tests/test_cli_integration.py
+        """
+
+        with tempfile.NamedTemporaryFile(mode="w", delete=False) as f:
+            f.write("test line 1\n")
+            f.write("another line\n")
+            f.write("test line 3\n")
+            test_file = f.name
+
+        try:
+            result = subprocess.run(
```

Ruff (0.14.1)

`subprocess` call: check for execution of untrusted input

(S603)